plugin-build/plugin/src/main/java/io/sentry/kotlin/multiplatform/gradle/plugin/SentryPlugin.kt

```diff
@@ -0,0 +1,231 @@
+    if (kmpExtension !is KotlinMultiplatformExtension) {
+        return
+    }
+
+    (kmpExtension as ExtensionAware).extensions.configure(CocoapodsExtension::class.java) { cocoapods ->
+        val podName = "Sentry"
+        val sentryPod = cocoapods.pods.findByName(podName)
+        if (sentryPod == null) {
+            cocoapods.pod(podName) {
+                version = cocoapodsAutoInstallExtension.sentryCocoaVersion.get()
+                linkOnly = true
+                extraOpts += listOf("-compiler-option", "-fmodules")
+            }
+        }
+    }
+}
+
+internal fun Project.configureLinkingOptions(linkerExtension: LinkerExtension) {
+    val kmpExtension = extensions.findByName(KOTLIN_EXTENSION_NAME)
+    if (kmpExtension !is KotlinMultiplatformExtension) {
+        // todo: log, not multiplatform found
+        return
+    }
+
+    val customXcodeprojPath = linkerExtension.xcodeprojPath.orNull
+    val derivedDataPath = findDerivedDataPath(customXcodeprojPath)
+
+    kmpExtension.appleTargets().all { target ->
+        val frameworkArchitecture = target.toSentryFrameworkArchitecture()
+        if (frameworkArchitecture == null) {
+            // todo: log, unsupported architecture
+            return@all
+        }
+
+        val dynamicFrameworkPath =
+            "$derivedDataPath/SourcePackages/artifacts/sentry-cocoa/Sentry-Dynamic/Sentry-Dynamic.xcframework/$frameworkArchitecture"
+        val staticFrameworkPath =
+            "$derivedDataPath/SourcePackages/artifacts/sentry-cocoa/Sentry/Sentry.xcframework/$frameworkArchitecture"
+
+        target.binaries.all binaries@{ binary ->
+            val path =
+                when {
+                    File(dynamicFrameworkPath).exists() -> dynamicFrameworkPath
+                    File(staticFrameworkPath).exists() -> {
+                        // todo: log, dynamic framework not found, try using static framework
+                        staticFrameworkPath
+                    }
+
+                    else -> {
+                        // todo: log, static framework also not found, error
+                        return@binaries
+                    }
+                }
+
+            if (binary is TestExecutable) {
+                binary.linkerOpts("-rpath", "$path/Sentry.framework", "-F$path")
+            } else if (binary is Framework) {
+                binary.linkerOpts("-F$path")
+            }
+        }
+    }
+}
+
+private fun Project.findDerivedDataPath(customXcodeprojPath: String? = null): String {
+    val xcodeprojPath = customXcodeprojPath ?: findXcodeprojFile(rootDir)?.absolutePath
+    val buildDirOutput = ByteArrayOutputStream()
+    exec {
+        it.commandLine =
+            listOf(
+                "xcodebuild", "-project", xcodeprojPath, "-showBuildSettings",
+            )
+        it.standardOutput = buildDirOutput
+    }
+    val buildSettings = buildDirOutput.toString("UTF-8")
+    val buildDirRegex = Regex("BUILD_DIR = (.+)")
```

Comment:
`m`: maybe worth to extract this into a companion object, so it's only computed once

Reply:
moved it as a top level declaration